binder/views.py

```diff
@@ -1158,10 +1169,51 @@ def _filter_relation(self, field_name, where_map, request, include_annotations):
 		return FilterDescription(q, need_distinct)
 
 
+	# handle self.alternative_filters as part of _parse_filter
+	def _parse_alternative_filters(self, field, filter_heads, *args, **kwargs):
+		head, tail = re.fullmatch(r'([^.:]*)(.*)', field).groups()
+
+		# split any/all of the tail, by default a filter is treated as any
+		#   NOTE: not is_any ==> is_all
+		is_any = bool(re.search(':any', tail))
+		is_all = bool(re.search(':all', tail))
+		is_not = bool(re.search(':not', tail))
+		if is_any and is_all:
+			raise BinderRequestError('Cannot use both :all and :any in alternative filter "{}".'.format(field))
+		# clean :all and :any from tail
+		tail = tail.replace(':all', '').replace(':any', '')
+
+		alts = []
+		for head in filter_heads:
+			field = head + tail
+			alt = self._parse_filter(field, *args, **kwargs)
+			alts.append(alt)
+		q, needs_distinct = alts[0]
+		for q_, needs_distinct_ in alts[1:]:
+			if not is_all and not is_not:
+				# :any (default)
+				# :not:all NOTE that not is maintained inside the filter
+				q |= q_
+			else:
+				# :all
+				# :not:any NOTE that not is maintained inside the filter
+				q &= q_
+			needs_distinct = needs_distinct or needs_distinct_
+		return FilterDescription(q, needs_distinct)
+
 	def _parse_filter(self, field, value, request, include_annotations, partial=''):
 		head, *tail = field.split('.')
 		need_distinct = False
 
+		try:
+			# get head without trailing :sorter
+			alt_head = head.split(':')[0]
```

Comment:
This is not the part where the `KeyError` can happen so it should be outside of the `try`, best practice is to keep the code in the `try` block as short as possible so that you really only catch the exception you want to catch.